src/Symfony/Component/Config/Tests/Util/XmlUtilsTest.php

```diff
@@ -55,13 +55,28 @@ public function testLoadFile()
             XmlUtils::loadFile($fixtures.'valid.xml', array($mock, 'validate'));
             $this->fail();
         } catch (\InvalidArgumentException $e) {
-            $this->assertContains('is not valid', $e->getMessage());
+            $this->assertRegExp('/The XML file "[\w:\/\\\.]+" is not valid\./', $e->getMessage());
         }
 
         $this->assertInstanceOf('DOMDocument', XmlUtils::loadFile($fixtures.'valid.xml', array($mock, 'validate')));
         $this->assertSame(array(), libxml_get_errors());
     }
 
+    public function testParse()
+    {
+        $fixtures = __DIR__.'/../Fixtures/Util/';
+
+        $mock = $this->getMockBuilder(__NAMESPACE__.'\Validator')->getMock();
+        $mock->expects($this->once())->method('validate')->will($this->onConsecutiveCalls(false, true));
+
+        try {
+            XmlUtils::parse(file_get_contents($fixtures.'valid.xml'), array($mock, 'validate'));
+            $this->fail();
```

Comment:
Other tests are written this way, because they try to test multiple cases in the same test. This is a legacy from tests written in the early days of Symfony's usage of PHPUnit.